CMakeLists.txt

```diff
@@ -136,6 +136,8 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     set (WARN_FLAGS "${WARN_FLAGS} -Wno-c++2a-compat -Wno-unknown-warning-option -Wno-suggest-override -Wno-suggest-destructor-override")
   elseif (CMAKE_HOST_APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "11.0")
     set (WARN_FLAGS "${WARN_FLAGS} -Wno-c++2a-compat")
+  elseif (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "10.0")
+    set (WARN_FLAGS "${WARN_FLAGS} -Wno-suggest-override -Wno-suggest-destructor-override -Wno-implicit-fallthrough")
```

Comment:
To avoid any code change in this test coverage PR, I simply excluded the existing errors.